libraries/AR_Motors/AP_MotorsUGV.cpp

```diff
@@ -266,11 +266,13 @@ float AP_MotorsUGV::get_slew_limited_throttle(float throttle, float dt) const
  */
 bool AP_MotorsUGV::have_skid_steering() const
 {
-    if (SRV_Channels::function_assigned(SRV_Channel::k_throttleLeft) &&
-        SRV_Channels::function_assigned(SRV_Channel::k_throttleRight)) {
-        return true;
-    }
-    return false;
+    bool simple_skid_steering = SRV_Channels::function_assigned(SRV_Channel::k_throttleLeft) && SRV_Channels::function_assigned(SRV_Channel::k_throttleRight);
+    bool omni = SRV_Channels::function_assigned(SRV_Channel::k_motor1) && 
+                SRV_Channels::function_assigned(SRV_Channel::k_motor2) &&
+                SRV_Channels::function_assigned(SRV_Channel::k_motor3) && 
+                SRV_Channels::function_assigned(SRV_Channel::k_motor4);
```

Comment:
We do support 3 motor omni, I think you could just check `_motors_num > 0`.

I did a little rework in this area in https://github.com/ArduPilot/ardupilot/pull/18612

Reply:
Thanks @IamPete1 for letting me know. Please check if I have done the right thing now.